scraper/cazy_webscraper.py

```diff
@@ -526,7 +504,7 @@ def parse_family_pages(family_url, family_name, cazy_home, logger):
             f"Incorrect formating of first protein table page URL: {first_pagination_url}\n"
             "Will not try and connect to this URL."
         )
-        return []
+        return family
```

Comment:
I think that failure to capture the correct data resulting in an error (here an `AttributeError`) should either raise a new informative error and propagate that upwards, or propagate the original error upwards. Here it appears that `try/except` is being used as flow control, and a `Family` instance with no data is being returned.

Options here might include throwing an error and skipping this family, or recording an error message in the returned `Family` instance to state that the URL could not be parsed. It might also be useful, in any case, to associate the base URL for a family with the `Family` instance by including it as an attribute.